web/src/components/organisms/Workspace/Settings/hooks.ts

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from "react";
+import { useNavigate } from "react-router-dom";
+
+import Notification from "@reearth-cms/components/atoms/Notification";
+import {
+  useUpdateWorkspaceMutation,
+  useDeleteWorkspaceMutation,
+} from "@reearth-cms/gql/graphql-client-api";
+import { useT } from "@reearth-cms/i18n";
+import { useWorkspace } from "@reearth-cms/state";
+
+export default () => {
+  const navigate = useNavigate();
+  const [currentWorkspace, setCurrentWorkspace] = useWorkspace();
+  const t = useT();
+
+  const workspaceId = currentWorkspace?.id;
+  const workspaceName = currentWorkspace?.name;
+
+  const [updateWorkspaceMutation] = useUpdateWorkspaceMutation();
+  const [deleteWorkspaceMutation] = useDeleteWorkspaceMutation({
+    refetchQueries: ["GetMe"],
+  });
```

Comment:
Does the updateWorkspaceMutation not also need to refetch?

Reply:
@KaWaite , no need, I'm updating the state of workspace in case of success.
In case of delete I need to remove the workspace from the list, that's why I refetch.

Reply:
That is fine for displaying the workspace settings page, but what about the workspace list in the header? Does the name there update if you don't refetch?

Reply:
@KaWaite , yes it is updated.